starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala

```diff
@@ -93,6 +88,8 @@ object SubmitPipeline extends DateTimeFunctions with WaitForIt with DateTimeMapp
 
         (getPeriodFromSchedule(specifiedSchedule), getStartTimeFromSchedule(specifiedSchedule))
       case x =>
+        val jarFile = S3FileHandler.getFileFromS3(opts.jar, opts.baseDir)
+        if (opts.cleanUp) jarFile.deleteOnExit()
```

Comment:
Is `deleteOnExit` asynchronous?

Reply:
Also: why would we ever not want to cleanup?

Reply:
local submit/testing

Reply:
Doc says

```
deleteOnExit()
Requests that the file or directory denoted by this abstract pathname be deleted when the virtual machine terminates.
```

Not sure if async or not